File: 09-DesignPatterns-Java/Solution/Observer/src/WeatherData.java
import java.util.ArrayList;
import java.util.List;

public class WeatherData extends WeatherAPI {
    private final List<IView> _observer = new ArrayList<>();

    public void Attach(IView view) {
        _observer.add(view);
    }

    public void Detach(IView view) {
        _observer.remove(view);
    }

    @Override
    protected void measurementChanged() {
        var temperature = getTemperature();
        var humidity = getHumidity();
        var windSpeed = getWindSpeed();
        var rainProbability = getRainProbability();

        for(var observer : _observer)
            observer.update(temperature, humidity, windSpeed, rainProbability);
    }
}

File: 10-DesignPatterns-CSharp/10-1-1-Observer-Solution/WeatherData.cs
namespace Observer
{
    using System.Collections.Generic;

    public sealed class WeatherData : WeatherAPI
    {
        private readonly List<IView> _observer = new();

        public void Attach(IView view) => _observer.Add(view);

        public void Detach(IView view) => _observer.Remove(view);

        protected override void MeasurementChanged()
        {
            var temperature = GetTemperature();
            var humidity = GetHumidity();
            var windSpeed = GetWindSpeed();
            var rainProbability = GetRainProbability();

            foreach (var observer in _observer)
                observer.Update(temperature, humidity, windSpeed, rainProbability);
        }
    }
}
